src/multi.jl

```diff
@@ -9,45 +9,131 @@ function amp_post_convhull(m::PODNonlinearModel; kwargs...)
+        if ((nl_type == :multilinear) || (nl_type == :bilinear)) && (m.nonlinear_terms[k][:convexified] == false)
+            λ, α = amp_convexify_multilinear(m, k, λ, α, discretization)
+        elseif nl_type == :monomial && !m.nonlinear_terms[k][:convexified]
+            λ, α = amp_convexify_monomial(m, k, λ, α, discretization)
+        elseif nl_type == :binlin && !m.nonlinear_terms[k][:convexified]
+            amp_convexify_binlin(m, k, discretization)
+        elseif nl_type == :binprod && !m.nonlinear_terms[k][:convexified]
+            β = amp_convexify_binprod(m, k, β)
         end
     end
 
     return
 end
 
+function amp_convexify_multilinear(m::PODNonlinearModel, k, λ::Dict, α::Dict, discretization::Dict)
+    m.nonlinear_terms[k][:convexified] = true  # Bookeeping the convexified terms
+
+    ml_indices, dim, extreme_point_cnt = amp_convhull_prepare(m, discretization, k)   # convert key to easy read mode
+    λ = amp_convhull_λ(m, k, ml_indices, λ, extreme_point_cnt, dim)
+    λ = populate_convhull_extreme_values(m, discretization, ml_indices, λ, dim, ones(Int,length(dim)))
+    α = amp_convhull_α(m, ml_indices, α, dim, discretization)
+    amp_post_convhull_constrs(m, λ, α, ml_indices, dim, extreme_point_cnt, discretization)
+
+    return λ, α
+end
+
+function amp_convexify_monomial(m::PODNonlinearModel, k, λ::Dict, α::Dict, discretization::Dict)
+    m.nonlinear_terms[k][:convexified] = true  # Bookeeping the convexified terms
+
+    monomial_index, dim, extreme_point_cnt = amp_convhull_prepare(m, discretization, k, monomial=true)
+    λ = amp_convhull_λ(m, k, monomial_index, λ, extreme_point_cnt, dim)
+    λ = populate_convhull_extreme_values(m, discretization, monomial_index, λ)
+    α = amp_convhull_α(m, [monomial_index], α, dim, discretization)
+    amp_post_convhull_constrs(m, λ, α, monomial_index, dim, discretization)
+
+    return λ, α
+end
+
+
+"""
+    TODO: docstring
+    This function redirect the bpml term to its right convexification
+"""
+function amp_convexify_binlin(m::PODNonlinearModel, k::Any, discretization::Dict)
+
+    m.nonlinear_terms[k][:convexified] = true  # Bookeeping the convexified terms
+
+    @assert length(m.nonlinear_terms[k][:var_idxs]) == 2
+    lift_idx = m.nonlinear_terms[k][:y_idx]
+    bin_idx = [i for i in m.nonlinear_terms[k][:var_idxs] if m.var_type_lifted[i] == :Bin]
+    cont_idx = [i for i in m.nonlinear_terms[k][:var_idxs] if m.var_type_lifted[i] == :Cont]
+
+    @assert length(bin_idx) == 1
+    @assert length(cont_idx) == 1
+    bin_idx = bin_idx[1]
+    cont_idx = cont_idx[1]
+
+    mccormick(m.model_mip, Variable(m.model_mip, lift_idx),
+        Variable(m.model_mip, bin_idx), Variable(m.model_mip, cont_idx),
+        0, 1, m.l_var_tight[cont_idx], m.u_var_tight[cont_idx])
+
+    return
+end
+
+"""
+    TODO: docstring
+    This function is very important
+"""
+function amp_convexify_binprod(m::PODNonlinearModel, k, β::Dict)
```

Comment:
This function is fine.

Reply:
The two functions do the same tasks with two different methods. One of it will be taken out later.